internal/indexer/indexer.go

```diff
@@ -309,22 +331,27 @@ func (i *Indexer) markRange(pos token.Position) bool {
 }
 
 // indexDefinition emits data for the given definition object.
-func (i *Indexer) indexDefinition(p *packages.Package, filename string, document *DocumentInfo, pos token.Position, obj types.Object) uint64 {
-	// Create a hover result vertex and cache the result identifier keyed by the definition location.
-	// Caching this gives us a big win for package documentation, which is likely to be large and is
-	// repeated at each import and selector within referenced files.
-	hoverResultID := i.makeCachedHoverResult(nil, obj, func() []protocol.MarkedString {
-		return findHoverContents(i.packageDataCache, i.packages, p, obj)
-	})
-
+func (i *Indexer) indexDefinition(p *packages.Package, filename string, document *DocumentInfo, pos token.Position, obj types.Object, typeSwitchHeader bool) uint64 {
 	rangeID := i.emitter.EmitRange(rangeForObject(obj, pos))
 	resultSetID := i.emitter.EmitResultSet()
 	defResultID := i.emitter.EmitDefinitionResult()
 
 	_ = i.emitter.EmitNext(rangeID, resultSetID)
 	_ = i.emitter.EmitTextDocumentDefinition(resultSetID, defResultID)
 	_ = i.emitter.EmitItem(defResultID, []uint64{rangeID}, document.DocumentID)
-	_ = i.emitter.EmitTextDocumentHover(resultSetID, hoverResultID)
+
+	if typeSwitchHeader {
+		// TODO(efritz) - not sure how to document a type switch header symbolic variable
+		// I'd like to somehow keep the type string of the RHS, but I'm not yet sure how
+		// to resolve that data given what we have.
```

Comment:
if you could give an example of what youre thinking here and how you envisage it being displayed to the user thatd be really appreciated so I've an idea whats going on :)

Reply:
I'm not sure yet, but I'd think something like:

```go
var v interface{} = 30

switch e := v.(type) {
// ...
}
```

Then e's hover text would be `var e interface{}` in the header.